faqs/too_long_without_yielding.md

```diff
@@ -0,0 +1,31 @@
+**Too Long Without Yielding**
+
+Computercraft has a builtin system which stops a single computer from running for too long at a time. After ~7 seconds of nonstop running, your computer will receive an error, "`Too long without yielding`."
+
+If, a few seconds after your program received the initial error, your program still has not yielded (ie: the error was caught by `pcall` or `xpcall`), the computer itself will shut down.
+
+**How do I fix this?**
+
+First, you need to figure out where your program is looping. The error itself may not be too helpful, as the `Too long without yielding` error points to the last line that was run before timing out (which could be OS code, or some other library's code). In most cases, you're looking for a `while true do` loop.
+
+Once you find where it's looping forever, you can simply add `sleep()` or `os.sleep()`. In most cases this will fix the problem. `sleep` has a call to a lua-builtin function, `coroutine.yield` which is what the error is referring to. `sleep` also has a bit of other event handling so that it can immediately resume.
```

Comment:
I know I'm late to the party, but I'd contend that this is not the first thing you should do. Most of the time an infinite loop is a bug, and when it's not you probably can listen to some event (for instance, `redstone`).

Reply:
Honestly I felt like putting in something extra about sleep being code-smell (especially in an event oriented environment) as well and to use `os.pullEvent`/others, but figured people who were asking about this error for the first time may not know how to adapt their code properly to use `os.pullEvent` rather than looping.